### PR TITLE
fix: fixed include flag KeyError

### PR DESCRIPTION
**Pull request by AKBrueggemann**
# Description

 There was a problem with the include_flag in common.smk. The get_assemblies_for_submission function gives the path to the faster file, not the name, leading to a KeyError in get_include_flag. This causes the inclusion of files in the high-quality-genomes, that should be excluded. Problem fixed by changing the functions used in get_include_flag_for_date and taking into account the lengths of the lists from get_assemblies_for_submission and get_sample_for_date.

## Checklist

- [x] I've formatted the PR title in accordance with the [structure of the conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I've updated the code style using [`pre-commit`](https://ikim-essen.github.io/uncovar/dev-guide/contributing/#pre-commit) if needed.
- [ ] I've updated or supplemented the documentation as needed.
- [x] I've read the [`CODE_OF_CONDUCT.md`] document.
- [x] I've read the [`CONTRIBUTING.md`] guide.

[`CODE_OF_CONDUCT.md`]: https://github.com/IKIM-Essen/uncovar/blob/master/CODE_OF_CONDUCT.md
[`CONTRIBUTING.md`]: https://github.com/IKIM-Essen/uncovar/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: workflow/rules/common.smk [hunk 1]
```diff
@@ -1272,10 +1272,23 @@ def get_include_flag(sample):
...
+    else:
+        allsamplelist = []
+        for sample in get_samples_for_date(wildcards.date):
+            for sample_path in get_assemblies_for_submission(
+                wildcards, "accepted samples"
+            ):
+                if sample in sample_path:
+                    allsamplelist.append(sample)
+                else:
+                    continue
```
Comment (thomasbtf) [inline workflow/rules/common.smk:1290]: I`m not sure if this is needed 🤔

**Review by thomasbtf — COMMENTED** [review]
Nice catch! Just one remark

**Review by thomasbtf — APPROVED** [review]

File: workflow/rules/common.smk (hunk 1 above)
Comment (alethomas) [inline workflow/rules/common.smk:1290]:
```suggestion
```

Comment (alethomas) [reply workflow/rules/common.smk:1290]: I'd also vote for leaving it out. If it's needed, we will see it 😉

**Review by alethomas — CHANGES_REQUESTED** [review]

**Review by alethomas — APPROVED** [review]

**Pull request merged**